src/backend/models.py

```diff
@@ -1,67 +1,95 @@
 from .exceptions import CannotBeRemovedException
 from django.db import models
 
+class Restaurant(models.Model):
+    name = models.CharField(max_length=30)
+    address = models.CharField(max_length=30)
+    phone_number = models.CharField(max_length=15)
+
+    def add_product(self, product_name: str, estimated_price: float):
+        product = Product(name=product_name, restaurant=self, estimated_price=estimated_price)
+        product.save()
+
+    def products_quantity(self):
+        return self.products.count()
```

Comment:
Solo a modo de comentario y por si nos interesa a futuro, en python podes declarar atributos decorando el método con @property, en este caso quedaría:

```suggestion
    @property
    def products_quantity(self):
        return self.products.count()
```
Con esto podes hacer `restaurant.products_quantity` y te devuelve la cuenta de los productos (sin usar los parentesis para llamar al metodo)